components/style/properties/longhand/ui.mako.rs

```diff
@@ -29,3 +29,38 @@ ${helpers.single_keyword("-moz-window-dragging", "default drag no-drag", product
                          gecko_enum_prefix="StyleWindowDragging",
                          animatable=False,
                          spec="None (Nonstandard Firefox-only property)")}
+
+<%helpers:longhand name="caret-color" animatable="False" products="none"
+                   spec="https://drafts.csswg.org/css-ui/#caret-color">
+    use std::fmt;
+    use style_traits::ToCss;
+    pub use self::computed_value::T as SpecifiedValue;
+
+    pub mod computed_value {
+        use cssparser::{Color, Parser};
+        use parser::{Parse, ParserContext};
+        use values::{Auto, Either};
+        use values::HasViewportPercentage;
+        use values::computed::ComputedValueAsSpecified;
+
+        no_viewport_percentage!(Color);
+        impl ComputedValueAsSpecified for Color {}
+
+        impl Parse for Color {
```

Comment:
Right, I tried using that previously but I was getting the error listed below when trying to run tests. Looking into it more I still can't really understand why the Rust compiler thinks that `computed` in this context is referring to a value of type `values::Either<cssparser::Color, values::Auto>` . At this point I've removed all references to `cssparser::Color` from my new code (including tests), and also ran `./mach clean`, but the build is still failing due to that error. Any ideas?

```
   Compiling style v0.0.1 (file:///Users/austinprete/rust/servo_dev/servo/components/style)
error[E0308]: mismatched types
     --> /Users/austinprete/rust/servo_dev/servo/target/debug/build/style-38f5dc3ba60f986e/out/properties.rs:44286:72
      |
44286 |                                                       .set_caret_color(computed );
      |                                                                        ^^^^^^^^ expected struct `values::specified::CSSColor`, found enum `csspars
er::Color`
      |
      = note: expected type `values::Either<values::specified::CSSColor, values::Auto>`
                 found type `values::Either<cssparser::Color, values::Auto>`

error: aborting due to previous error

error: Could not compile `style`.
```